clippy_lints/src/manual_empty_string_creation.rs

```diff
@@ -0,0 +1,141 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use rustc_ast::LitKind;
+use rustc_errors::Applicability::Unspecified;
+use rustc_hir::{Expr, ExprKind, PathSegment, QPath, TyKind};
+use rustc_lint::{LateContext, LateLintPass, LintContext};
+use rustc_middle::{lint::in_external_macro, ty};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::{sym, symbol, Span};
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// Checks for usage of `""` to create a `String`, such as `"".to_string()`, `"".to_owned()`,
+    /// `String::from("")` and others.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Different ways of creating an empty string makes your code less standardized, which can
+    /// be confusing.
+    ///
+    /// ### Example
+    /// ```rust
+    /// let a = "".to_string();
+    /// let b: String = "".into();
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// let a = String::new();
+    /// let b = String::new();
+    /// ```
+    #[clippy::version = "1.64.0"]
+    pub MANUAL_EMPTY_STRING_CREATION,
+    style,
+    "empty String is being created manually"
+}
+declare_lint_pass!(ManualEmptyStringCreation => [MANUAL_EMPTY_STRING_CREATION]);
+
+impl LateLintPass<'_> for ManualEmptyStringCreation {
+    fn check_expr(&mut self, cx: &LateContext<'_>, expr: &Expr<'_>) {
+        if in_external_macro(cx.sess(), expr.span) {
+            return;
+        }
+
+        let ty = cx.typeck_results().expr_ty(expr);
+        match ty.kind() {
+            ty::Adt(adt_def, _) if adt_def.is_struct() => {
+                if !cx.tcx.is_diagnostic_item(sym::String, adt_def.did()) {
+                    return;
+                }
+            },
+            _ => return,
+        }
+
+        match expr.kind {
+            ExprKind::Call(func, args) => {
+                parse_call(cx, expr.span, func, args);
+            },
+            ExprKind::MethodCall(path_segment, args, _) => {
+                parse_method_call(cx, expr.span, path_segment, args);
+            },
+            _ => (),
+        }
+    }
+}
+
+/// Checks if an expression's kind corresponds to an empty &str.
+fn is_expr_kind_empty_str(expr_kind: &ExprKind<'_>) -> bool {
+    if  let ExprKind::Lit(lit) = expr_kind &&
+        let LitKind::Str(value, _) = lit.node &&
+        value == symbol::kw::Empty
+    {
+        return true;
+    }
+
+    false
+}
+
+/// Emits the `MANUAL_EMPTY_STRING_CREATION` warning and suggests the appropriate fix.
+fn warn_then_suggest(cx: &LateContext<'_>, span: Span) {
+    span_lint_and_sugg(
+        cx,
+        MANUAL_EMPTY_STRING_CREATION,
+        span,
+        "empty String is being created manually",
+        "consider using",
+        "String::new()".into(),
+        Unspecified,
```

Comment:
I was a little bit unsure if this new lint can have false positives. Do you think it seems safe to make it `MachineApplicable`? 

It worked fine in clippy's codebase, but since I'm still unexperienced with these lints I'm not sure.

Reply:
I don't see how the suggestions can go really wrong with this lint. Making it machine applicable seems fine, the lint will still trigger in FP cases either way.

Reply:
Thanks for the suggestions!  Fixed in 4e9aebfbd2b9b94b08d27f0d61ea47d8e9e80633